fields/sql.go

```diff
@@ -97,3 +100,157 @@ func (f Valuer) Value() (driver.Value, error) {
 
 var _ driver.Valuer = Valuer{}
 var _ driver.Valuer = &Valuer{}
+
+// Scanner fulfills the sql.Scanner interface which deserializes SQL values
+// into the type dictated by our descriptor.
+type Scanner struct {
+	*Descriptor
+	value   reflect.Value
+	IsValid bool
```

Comment:
does `IsValid` need to be persisted on the struct? I see that it is only referenced within `Scan`

Reply:
Yep, since we treat the value as a non-pointer value, IsValid tells us whether or not something is nil

Reply:
could you point to where that is used outside of `Scan`?

Reply:
ah, thanks i missed that. making it package private sounds good